Sources/Common/Util/KeyValueStorage.swift

```diff
@@ -20,67 +22,84 @@ public protocol KeyValueStorage: AutoMockable {
  */
 // sourcery: InjectRegister = "KeyValueStorage"
 public class UserDefaultsKeyValueStorage: KeyValueStorage {
-    private let siteId: String
+    private var siteId: String?
+    private let deviceMetricsGrabber: DeviceMetricsGrabber
+
+    private var userDefaults: UserDefaults? {
+        UserDefaults(suiteName: getFileName())
+    }
 
-    init(siteId: SiteId) {
+    init(siteId: SiteId, deviceMetricsGrabber: DeviceMetricsGrabber) {
         self.siteId = siteId
+        self.deviceMetricsGrabber = deviceMetricsGrabber
+    }
+
+    // Used for global data that's relevant to *all* of the site-ids (not sandboxed).
+    // Instead of the more common way the SDK stores data by sandboxing all of that data by site-id.
+    public func switchToGlobalDataStore() {
+        siteId = nil
     }
 
     /**
-     We want to separate all of the data for each CIO workspace and app.
+     We want to sandbox all of the data for each CIO workspace and app.
      Therefore, we use the app's bundle ID to be unique to the app and the siteId to separate all of the sites.
 
      We also need to have 1 set of UserPreferences that all workspaces of an app share.
-     For these moments, use `sharedSiteId` as the `siteId` parameter.
+     For these moments, use `shared` as the `siteId` value.
      */
-    private func getUserDefaults() -> UserDefaults? {
+    internal func getFileName() -> String {
         var appUniqueIdentifier = ""
-        if let appBundleId = DeviceMetricsGrabber.appBundleId {
+        if let appBundleId = deviceMetricsGrabber.appBundleId {
             appUniqueIdentifier = ".\(appBundleId)"
         }
 
-        return UserDefaults(suiteName: "io.customer.sdk\(appUniqueIdentifier).\(siteId)")
+        var siteIdPart = ".shared" // used for storing global data used for all site-ids.
+        if let siteId = siteId { // if a siteid is given to this instance, we dont store global data with this instance.
+            siteIdPart = ".\(siteId)"
+        }
+
+        return "io.customer.sdk\(appUniqueIdentifier)\(siteIdPart)"
```

Comment:
Reminder to self: QA test this to make sure that we are *not* changing the name after customer upgrades. To avoid customers losing data.